server/src/services/metadata.service.ts

```diff
@@ -334,6 +336,29 @@ export class MetadataService extends BaseService {
     return JobStatus.SUCCESS;
   }
 
+  private getImageDimensions(exifTags: ImmichTags): { width: number; height: number } {
+    /*
+     * The "true" values for width and height are a bit hidden, depending on the camera model and file format.
+     * For RAW images in the CR2 or RAF format, the "ImageSize" value seems to be correct,
+     * but ImageWidth and ImageHeight are not correct (they contain the dimensions of the preview image).
+     */
+
+    let width = Number.NaN;
+    let height = Number.NaN;
+    const imageSize = exifTags.ImageSize;
+    if (imageSize && imageSize.indexOf('x') > 0) {
+      // width x height (e.g. 100x200)
+      const split = imageSize.split('x');
+      width = Number.parseInt(split[0]);
+      height = Number.parseInt(split[1]);
+    }
+    if (!width || !height) {
+      width = exifTags.ImageWidth ?? Number.NaN;
+      height = exifTags.ImageHeight ?? Number.NaN;
+    }
+    return { width, height };
```

Comment:
```suggestion
    let [width, height] = exifTags.ImageSize?.split('x') || [];
    if (!width || !height) {
      [width, height] = [exifTags.ImageWidth, exifTags.ImageHeight];
    }
    return { width: Number.parseInt(width), height: Number.parseInt(height) };
```

Reply:
That won't work due to different types. I'd rather have long code that's readable than short code that's a mystery. This isn't perl, after all.

Reply:
I wouldn't consider Mert's suggestion mystery. I think it's a pretty elegant and idiomatic solution that's very well readable. If you need to juggle around with types however I can see that this becomes less pretty.

Reply:
Well, I do. It doesn't work and uses typecasts (from number to string?). To me, that's quite a mystery.

Reply:
The cast is to cast `string | undefined` to string. The typing for `parseInt` only accepts string, but an undefined input results in NaN. You can avoid the type cast by changing `as string` to `|| ''`.

Reply:
Revised suggestion that handles typing more elegantly:
```suggestion
    let [width, height] = exifTags.ImageSize?.split('x').map((dim) => Number.parseInt(dim)) || [];
    if (!width || !height) {
      [width, height] = [exifTags.ImageWidth, exifTags.ImageHeight];
    }
    return { width, height };
```

Reply:
That still doesn't work.

Reply:
Maybe it's complex enough to prefer longer code that actually works (without employing elegant/mysterious code)? Just a suggestion.

Reply:
Can you clarify what doesn't work? I ran both functions in a REPL and they behave the same (except the suggested code uses `undefined` instead of `NaN`). The only change I needed to make for TS is adding a type hint for `dim` (quite normal).

Regardless, this is just a suggestion: you can keep the current code since this is what you prefer. I just think the suggestion is cleaner and more readable for something that (imo) isn't particularly complex.

Reply:
It doesn't compile due to type issues. I realize that this is a suggestion, but I prefer suggestions that actually work. I see that your code is shorter and possibly also cleaner, but that won't help if it doesn't work. If you're unsure of how to provide working suggestions, I suggest in return to try them out yourself before.